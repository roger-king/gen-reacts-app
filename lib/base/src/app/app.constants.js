//# sourceMappingURL=app.constants.js.map