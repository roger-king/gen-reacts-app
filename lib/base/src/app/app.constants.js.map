{"version":3,"file":"app.constants.js","sourceRoot":"","sources":["../../../../src/base/src/app/app.constants.ts"],"names":[],"mappings":""}