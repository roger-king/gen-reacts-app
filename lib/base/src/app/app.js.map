{"version":3,"file":"app.js","sourceRoot":"","sources":["../../../../src/base/src/app/app.tsx"],"names":[],"mappings":";;;;;;;;;;;;AAAA,6BAA+B;AAC/B,2CAA+C;AAE/C;IAAyB,uBAAyB;IAC9C,aAAY,KAAU;eAClB,kBAAM,KAAK,CAAC;IAChB,CAAC;IAEM,oBAAM,GAAb;QACI,OAAO,CAAC,4BAAe,GAAG,CAAC;IAC/B,CAAC;IACL,UAAC;AAAD,CAAC,AARD,CAAyB,KAAK,CAAC,SAAS,GAQvC;AARY,kBAAG"}