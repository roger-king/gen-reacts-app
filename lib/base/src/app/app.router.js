"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var app_theme_1 = require("./app.theme");
var react_router_dom_1 = require("react-router-dom");
var pages_1 = require("./pages");
var NoMatch = function (props) { return <div className={props.className}>404 Page Not Found</div>; };
var StyledNoMatch = app_theme_1.default(NoMatch)(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n    margin: 0 auto;\n    color: red;\n"], ["\n    margin: 0 auto;\n    color: red;\n"])));
exports.RouterContainer = function (props) { return (<react_router_dom_1.BrowserRouter>
        <react_router_dom_1.Switch>
            <react_router_dom_1.Route path="/" component={pages_1.LoadableHomePage}/>
            <react_router_dom_1.Route component={StyledNoMatch}/>
        </react_router_dom_1.Switch>
    </react_router_dom_1.BrowserRouter>); };
var templateObject_1;
//# sourceMappingURL=app.router.js.map