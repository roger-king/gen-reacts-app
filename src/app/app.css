@lost flexbox flex;

html,
body {
    margin: 0 auto;
    min-height: 100%;
    min-width: 1024px;
    width: 100%;
    height: auto;
    overflow-x: hidden;
}

/* Global Fonts */
.title {
    font-family: 'Varela Round', sans-serif;
    font-size: 4em;
    display: block;
}

.content-font {
    font-family: 'Open Sans', sans-serif;
    font-size: 1em;
}

/* Global Inputs */
.arcing-input{
    position: relative;
    border-radius: 5px;
    height: 40px;
    width: 500px;
    margin: 10px;
}

.arcing-btn{
    position: relative;
    background-color: #007acc;
    border-radius: 5px;
    color: white;
    height: 40px;
    width: 80px;
    margin: 10px;
}
