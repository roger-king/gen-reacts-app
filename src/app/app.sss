html,
body {
    margin: 0 auto;
    /* Set rules to fill background */
    min-height: 100%;
    min-width: 1024px;
    /* Set up proportionate scaling */
    width: 100%;
    height: auto;
    /* Set up positioning */
    position: fixed;
}

.text-center {
    text-align: center;
}

.text-right {
    text-align: right;
}

.text-left {
    text-align: left;
}

.centered {
    margin: 0 auto;
}