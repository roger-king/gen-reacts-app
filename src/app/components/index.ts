/**
 * Global imports of components.
 * This helps for cleaner imports through out the application.
 */

// Global imports of all components (do not remove - will break automation!)
