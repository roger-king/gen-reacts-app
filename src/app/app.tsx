import * as React from 'react';
import DevTools, {configureDevtool} from 'mobx-react-devtools';
import {UIRouter, UIView} from '@uirouter/react';
import {states, plugins} from './router.config';
import 'antd/dist/antd.css';
import './app.scss';

/**
 * Main application container.
 * This wraps around the entire application
 * You can set things like app-wide css grid here.
 * 
 * @preferred
 */
export class App extends React.Component<{}, {}> {
    constructor() {
        super();
    }

    render() {
        return (
            <UIRouter plugins={plugins} states={states}>
                <div className="app-container">
                    <UIView/>
                </div>
            </UIRouter>
        );
    }
}
