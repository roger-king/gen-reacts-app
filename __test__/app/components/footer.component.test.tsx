import React from 'react';
import renderer from 'react-test-renderer';
import { Footer } from './../../../src/app/components';

let tree;

beforeEach(function () {
    tree = renderer.create(
        <Footer/>
    );
});

describe('<Footer/>', function () {
    test('renders properly', () => {
        let component = tree.toJSON();
        expect(component.children[0].type).toEqual('a');
        expect(component.children[0].props.href).toEqual('https://github.com/roger-king/react-typescript-gen');
        expect(component.children[0].props.target).toEqual('blank');
        expect(component.children[0].children[0].type).toEqual('img');
        expect(component.children[0].children[0].props.src).toEqual('static/images/github.png');
        expect(component.children[0].children[0].props.width).toEqual(50);
        expect(component.children[0].children[0].props.height).toEqual(50);
    });

    test('renders from snapshot', () => {
        expect(tree).toMatchSnapshot();
    });
});
