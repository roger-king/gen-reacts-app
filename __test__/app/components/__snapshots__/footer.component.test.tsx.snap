// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> renders from snapshot 1`] = `
<div
  className="footer-container"
>
  <a
    href="https://github.com/roger-king/react-typescript-gen"
    target="blank"
  >
    <img
      className="icon-container"
      height={50}
      src="static/images/github.png"
      width={50}
    />
  </a>
</div>
`;
